mne/viz/_3d.py

```diff
@@ -777,7 +786,95 @@ def _limits_to_control_points(clim, stc_data, colormap):
+
+    return scale_pts, colormap, transparent
+
+
+def _scale_mayavi_lut(lut_manager, fmin, fmid, fmax, transparent,
+                      verbose=None):
+        """Scale a mayavi colormap LUT to a given fmin, fmid and fmax.
+
+        This function operates on a Mayavi LUTManager. This manager can be
+        obtained through the traits interface of mayavi. For example:
+        `x.module_manager.vector_lut_manager`.
+
+        Parameters
+        ----------
+        lut : LUTManager
+            The Mayavi lookup table manager used to scale the colormap.
+        fmin : float
+            minimum value of colormap.
+        fmid : float
+            value corresponding to color midpoint.
+        fmax : float
+            maximum value for colormap.
+        transparent : boolean
+            if True: use a linear transparency between fmin and fmid.
+        verbose : bool, str, int, or None
+            If not None, override default verbose level (see mne.verbose).
+
+        Returns
+        -------
+        lut_table : 2D array (n_colors, 4)
+            The re-scaled color lookup table
+
+        Notes
+        -----
+        Much of this code was taken from PySurfer's scale_data_colormap
+        function.
+        """
+        if not (fmin < fmid) and (fmid < fmax):
+            raise ValueError("Invalid colormap, we need fmin<fmid<fmax")
+
+        # Cast inputs to float to prevent integer division
+        fmin = float(fmin)
+        fmid = float(fmid)
+        fmax = float(fmax)
+
+        logger.info("colormap: fmin=%0.2e fmid=%0.2e fmax=%0.2e "
+                    "transparent=%d" % (fmin, fmid, fmax, transparent))
+
+        # Get the original colormap
+        lut_table = lut_manager.lut.table.to_array()
+
+        # Add transparency if needed
+        if transparent:
+            n_colors = lut_table.shape[0]
+            n_colors2 = int(n_colors / 2)
+            lut_table[:n_colors2, -1] = np.linspace(0, 255, n_colors2)
+            lut_table[n_colors2:, -1] = 255 * np.ones(n_colors - n_colors2)
+
+        # Scale the colormap
+        lut_table_new = lut_table.copy()
+        n_colors = lut_table.shape[0]
+        n_colors2 = int(n_colors / 2)
+
+        # Index of fmid in new colorbar
+        fmid_idx = int(np.round(n_colors * ((fmid - fmin) /
+                                            (fmax - fmin))) - 1)
+
+        # Go through channels
+        for i in range(4):
+            part1 = np.interp(np.linspace(0, n_colors2 - 1, fmid_idx + 1),
+                              np.arange(n_colors),
+                              lut_table[:, i])
+            lut_table_new[:fmid_idx + 1, i] = part1
+            part2 = np.interp(np.linspace(n_colors2, n_colors - 1,
+                                          n_colors - fmid_idx - 1),
+                              np.arange(n_colors),
+                              lut_table[:, i])
+            lut_table_new[fmid_idx + 1:, i] = part2
+
+        return lut_table_new
```

Comment:
BTW I usually don't get my colorbar to look right when using MNE colormaps with nonlinear spacing out from zero (e.g., fmin=4, fmid=5, fmax=6), does this by any chance fix that?